pkg/summarizer/summary.go

```diff
@@ -156,21 +156,25 @@ func summaryPath(name string) string {
 	return "summary-" + normalizer.ReplaceAllString(strings.ToLower(name), "")
 }
 
-func writeSummary(ctx context.Context, client *storage.Client, path gcs.Path, sum *summarypb.DashboardSummary) error {
+func writeSummary(ctx context.Context, client gcs.Client, path gcs.Path, sum *summarypb.DashboardSummary) error {
 	buf, err := proto.Marshal(sum)
 	if err != nil {
 		return fmt.Errorf("marshal: %v", err)
 	}
-	return gcs.Upload(ctx, client, path, buf, gcs.DefaultAcl, "no-cache") // TODO(fejta): configurable cache value
+	return client.Upload(ctx, path, buf, gcs.DefaultAcl, "no-cache") // TODO(fejta): configurable cache value
 }
 
 // pathReader returns a reader for the specified path and last modified, generation metadata.
-func pathReader(ctx context.Context, client *storage.Client, path gcs.Path) (io.ReadCloser, time.Time, int64, error) {
-	r, err := client.Bucket(path.Bucket()).Object(path.Object()).NewReader(ctx)
+func pathReader(ctx context.Context, client gcs.Client, path gcs.Path) (io.ReadCloser, time.Time, int64, error) {
+	r, err := client.Open(ctx, path)
 	if err != nil {
-		return nil, time.Time{}, 0, fmt.Errorf("read %s: %w", path, err)
+		return nil, time.Time{}, 0, fmt.Errorf("client.Open(%s): %w", path, err)
```

Comment:
FYI I'd remove path from the returned error as it is trivial to the caller (aka the caller passes in `pathReader(..., path)` and therefore knows _exactly_ what it is and can decide whether or not to log/wrap it.

OTOH when the code is
```golang
for _, path := range paths {
  r, err := Open(path)
  if err != nil {
    return fmt.Errorf("Open(%q): %w", path, err)
  }
}
```

I _would_ include it because which of the many paths caused the problem is useful info.

Reply:
Good point, thanks! Done.